sys/trickle/trickle.c

```diff
@@ -51,7 +51,7 @@ void trickle_interval(trickle_t *trickle)
     /* old_interval == trickle->I / 2 */
     trickle->t = random_uint32_range(old_interval, trickle->I);
 
-    trickle->msg_time = (trickle->t + diff) * MS_PER_SEC;
+    trickle->msg_time = (trickle->t + diff) * US_PER_MS;
```

Comment:
I agree that from the documentation, `trickle->msg_time` is in `ms`, and that `xtimer_set_msg64` `offset` is in `µs`. However in that case it should be multiplied by `USEC_PER_MS` only for the `xtimer_set_msg64` call not when setting `trickle->msg_time`.

I do not really get what is the the `t + diff` unit is however. From what I see by doing grep, nothing else is using `msg_time` in the repository at the moment.

Reply:
Maybe @cgundogan can clarify why the `msg_time` is a member of the struct.

I agree that from the documentation, `msg_time` should be in ms, I can take another look to see if there are more units that can be clarified here.

Reply:
> why souldn't msg_time be a member of the struct? If several applications use trickle concurrently, then a distinct msg_time context is needed for all of them

Because it is only used in the `trickle_interval` interval function and only after the new value is calculated from the trickle struct values.

Reply:
>I do not really get what is the the t + diff unit is however.

it's in `ms`.

Reply:
@bergzand ah now I see! Yeah, in a previous iteration we used to timers to trigger the "new interval phase" and the "trigger callback phase". I would be fine removing that from the struct.